librtasevent_src/rtas_srcfru.c

```diff
@@ -260,6 +261,7 @@ parse_src_scn(struct rtas_event *re)
         if (fru == NULL) {
             cleanup_rtas_event(re);
             errno = ENOMEM;
+            free(src_raw);
```

Comment:
I think you can free `src_raw` before entering do-while loop (around line 250). That way you don't need to add free() call in all error path etc.

-Vasant

Reply:
@hegdevasant , I have improved the patch according to your suggestion.